File: sample/src/main/java/io/erva/sample/multi/MultiChoiceCell.java
package io.erva.sample.multi;

import android.view.View;
import android.widget.CheckBox;

import butterknife.BindView;
import butterknife.OnClick;
import io.erva.celladapter.Cell;
import io.erva.celladapter.Layout;
import io.erva.sample.BaseSelectableCell;
import io.erva.sample.R;

@Layout(R.layout.item_multi)
public class MultiChoiceCell extends BaseSelectableCell<MultiChoiceModel, Cell.Listener<MultiChoiceModel>> {

    @BindView(R.id.cb_multi) CheckBox checkBox;

    public MultiChoiceCell(View view) {
        super(view);
    }

    @Override
    protected void bindView() {
        checkBox.setText(getItem().multiTitle);
        checkBox.setChecked(selectionManager.isSelected(getAdapterPosition()));
    }

    @OnClick(R.id.cb_multi)
    void checkBoxClicked() {
        selectionManager.toggleSelection(getAdapterPosition());
        getListener().onCellClicked(getItem());
    }
}

File: sample-v7-kotlin/src/main/kotlin/io/erva/sample/multi/MultiChoiceCell.kt
package io.erva.sample.multi

import android.view.View
import io.erva.celladapter.Layout
import io.erva.celladapter.v7.Cell
import io.erva.celladapter.v7.select.SelectableCell
import io.erva.sample.R
import kotlinx.android.synthetic.main.item_multi.view.*

@Layout(R.layout.item_multi)
class MultiChoiceCell(view: View) : SelectableCell<MultiChoiceModel, Cell.Listener<MultiChoiceModel>>(view) {

    override fun bindView() {
        view.cb_multi.text = item().multiTitle
        view.cb_multi.isChecked = selectionManager.isSelected(adapterPosition)
        view.cb_multi.setOnClickListener {
            selectionManager.toggleSelection(adapterPosition)
            listener()?.onCellClicked(item())
        }
    }
}